RecoLocalFastTime/FTLClusterizer/plugins/MTDCPEESProducer.cc

```diff
@@ -31,7 +32,7 @@ class MTDCPEESProducer : public edm::ESProducer {
 
 MTDCPEESProducer::MTDCPEESProducer(const edm::ParameterSet& p) {
   pset_ = p;
-  setWhatProduced(this, "MTDCPEBase").setConsumes(ddToken_);
+  setWhatProduced(this, "MTDCPEFromSiPMTime").setConsumes(ddToken_);
```

Comment:
I'm confused about why this seemingly generic producer is now using a specific implementation.

Reply:
Hi,

The only difference between the "MTDCPEBase" and "MTDCPEFromSiPMTime" is just that "MTDCPEFromSiPMTime" added a more detailed position error information for MTD BTL. I think it's not more specific but just included position error more precisely. The reason I didn't change the "MTDCPEBase" is that I am not sure whether there are some other modules that are also using this class so changing it directly might cause some problems.

Thanks.

Reply:
@pmeridian @casarsa @fabiocos @parbol please comment on this. I can see two cases:
1. MTDCPEFromSiPMTime is the only algorithm that will be used, therefore it should be combined with or completely replace MTDCPEBase.
2. Multiple algorithms can be used, in which case a factory or similar pattern should be introduced to handle them.